crates/uv-interpreter/src/discovery.rs

```diff
@@ -62,8 +62,10 @@ pub enum VersionRequest {
 /// The policy for discovery of "system" Python interpreters.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
 pub enum SystemPython {
-    /// Do not allow a system Python
+    /// Only allow a system Python if passed directly i.e. via [`InterpreterSource::ProvidedPath`] or [`InterpreterSource::ParentInterpreter`]
     #[default]
+    Explicit,
+    /// Do not allow a system Python
     Disallowed,
```

Comment:
I decided to add an `Explicit` variant to avoid confusing internal behavior where `SystemPython::Disallowed` could still get a system Python from this environment variable.